meson.build

```diff
@@ -558,20 +558,10 @@ else
   charset_dep = []
 endif
 
-rl_req = get_option('readline')
-if rl_req != 'false'
-  readline_dep = c_compiler.find_library('readline', dirs: cross_lib_path,
-                                         required:false)
-
-  if readline_dep.found() and c_compiler.has_function('rl_completion_suppress_append',
-                                                      dependencies: readline_dep,
-                                                      args: ['-O'])
+# Readline 4.3 ensures that we have `rl_completion_suppress_append`
+readline_dep = dependency('readline', version: '>= 4.3', required: get_option('readline'))
```

Comment:
Simplest approach here: use feature options with find_library. Yes, that works. And you can defer migrating to dependency(), until another PR.

That was the original idea, right? Incremental fixes, separately reviewable? Don't get this PR bogged down on debating supported readlines.